### ISSUE TITLE
Location and all weather data "N/A"

### ISSUE THREAD

**Issue by quequotion** [casasfernando/wingpanel-indicator-weather#8]
I recently packaged this indicator [for the AUR](https://aur.archlinux.org/packages/wingpanel-indicator-weather-git).

The indicator appears in wingpanel, but none of the weather data, nor my location, are displayed.

Whether I use location discovery, or specify a location manually, nothing changes.

In dconf, I found that the indicator *is* aware of the location I specified, which is set in the key `com.github.casasfernando.wingpanel-indicator-weather.location-manual`.

I tried specifying various locations (Toyko, New York), but nothing gets results.

**Comment by casasfernando** [comment]
I saw your comment https://github.com/casasfernando/wingpanel-indicator-weather/issues/3#issuecomment-1045971747 but as far as I know it shouldn't be a problem.

I just try to configure Tokyo, Japan as my location and it's working for me as you can see below:

![image](https://user-images.githubusercontent.com/15861819/154835978-724ca23a-93de-435c-983f-7bd4b06bcfef.png)

![image](https://user-images.githubusercontent.com/15861819/154836007-5fcb0ace-255e-4ff0-a875-59b80501f171.png)

You can try a couple of things to try to troubleshoot this issue:

1. When manually selecting a location, as you type the name you will see a list of suggested locations, click on the name of the location in the list of suggestions to autocomplete and update the location in the indicator settings and let me know if that works. Please bear in mind that if there are no suggestions for the location name you are entering is because the location is not in libgweather locations database, hence you will have to try names of different locations that are near to the one you are looking for until you find one that is in database. More information: https://wiki.gnome.org/Projects/LibGWeather/ImprovingLocations

2. Start collecting a wireshark network capture. Click on the "Update Weather Data" option in the indicator. Wait for 10-20 seconds and end the capture. If you go through the capture you should see a http request to grab weather data from one of the weather services available (mainly US weather office -IWIN-).

Let me know if the above works.

**Comment by gingerzx** [comment]
Same problem i checked my city was in the list, also tried wireshark could not get a result either. Its still N/A and it does not saves the location. When i open the settings again my city does not show up.

**Comment by dmartind72** [comment]
Hola! Como estas? Estoy probando tu aplicacion en Ultramarine Pantheon Desktop, es de Fedora version 35, y no me funciona.  Si bien hice lo que indicaste (actualice libgweather, libgeoclue desde las paginas de gnome y gitlabs) mas arriba igualmente sigue sin funcionar. Adjunto una imagen de como se ve en mi escritorio. Si podrias ver cual o cuales son las razones porque no funciona te lo agradeceria mucho.  Muchas gracias. Cordiales saludos. 

[Captura realizada el 2022-04-16 22 16 30](https://user-images.githubusercontent.com/103865887/163696175-51a75961-8f9c-425d-9122-22ff26f36703.png)

**Comment by casasfernando** [comment]
> Same problem i checked my city was in the list, also tried wireshark could not get a result either. Its still N/A and it does not saves the location. When i open the settings again my city does not show up.

Can you share more information please?

1- Are you running on elementary OS or another distro? Which version?
2- Are you installing from source or from the deb package available in the release page of the project?
3- Can you collect a log file so I can take a look at it?
    a- Open a Terminal window and terminate any wingpanel process (you may need to do this twice).
    b- *Make sure no wingpanel process is running* and start a new wingpanel instance manually by running: `G_MESSAGES_DEBUG=all io.elementary.wingpanel` for elementary OS 6.x or `G_MESSAGES_DEBUG=all wingpanel` for elementary OS 5.x
    c- Reproduce the issue (e.g.: click on the Update Weather Data button to refresh weather data or select a new location in the indicator settings)
    d- Save the output from the Terminal window to a text file and share it over here.

Thanks

**Comment by casasfernando** [comment]
> Hola! Como estas? Estoy probando tu aplicacion en Ultramarine Pantheon Desktop, es de Fedora version 35, y no me funciona. Si bien hice lo que indicaste (actualice libgweather, libgeoclue desde las paginas de gnome y gitlabs) mas arriba igualmente sigue sin funcionar. Adjunto una imagen de como se ve en mi escritorio. Si podrias ver cual o cuales son las razones porque no funciona te lo agradeceria mucho. Muchas gracias. Cordiales saludos.
> 
> [Captura realizada el 2022-04-16 22 16 30](https://user-images.githubusercontent.com/103865887/163696175-51a75961-8f9c-425d-9122-22ff26f36703.png)

Your case is more tricky. *This indicator was written for and tested on elementary OS only*.
I'm not planning to support other distributions at least for now but if you or anyone else wants to try it on different distros, debug the issues you may find and contribute back with code or tips to make it work it will be more than welcome.

From the screenshot you provided I can see that you are missing the icons to start with, so that will be a problem that needs to be fixed. Weather status icons and others are coming directly from the distro and are not distributed with the indicator.

You can follow step 3 from https://github.com/casasfernando/wingpanel-indicator-weather/issues/8#issuecomment-1116683005 to try to troubleshoot/debug the issue(s) you are having.

Thanks

**Comment by DeeeeLAN** [comment]
```
(io.elementary.wingpanel:30337): dconf-DEBUG: 03:46:33.408: change_fast

(io.elementary.wingpanel:30337): dconf-DEBUG: 03:46:33.408: change_notify: /com/github/casasfernando/wingpanel-indicator-weather/weather-last-update-req



(io.elementary.wingpanel:30337): GWeather-CRITICAL **: 03:46:33.408: gweather_info_update: assertion 'info->contact_info != NULL' failed
```

Here are the relevant lines that occurred when I selected my location manually.

**Comment by casasfernando** [comment]
> ```
> (io.elementary.wingpanel:30337): dconf-DEBUG: 03:46:33.408: change_fast
> (io.elementary.wingpanel:30337): dconf-DEBUG: 03:46:33.408: change_notify: /com/github/casasfernando/wingpanel-indicator-weather/weather-last-update-req
> 
> (io.elementary.wingpanel:30337): GWeather-CRITICAL **: 03:46:33.408: gweather_info_update: assertion 'info->contact_info != NULL' failed
> ```
> 
> Here are the relevant lines that occurred when I selected my location manually.

Thanks for the log excerpt I will look into it.
Would you mind sharing more details about your platform? Like distro, version, libgweather version, etc.?

**Comment by DeeeeLAN** [comment]
Sure, I am running Arch, with Pantheon installed for my desktop environment. Here are the versions I have installed:
```
wingpanel-indicator-weather-git r139.62ba375-1
libgweather 40.0+r87+g80e5a652-2
libgweather-4 4.0.0-1
```
I am not sure which libgweather you are using.

**Comment by DeeeeLAN** [comment]
I just downgraded libgweather to
```
libgweather 3.36.1-2
```
And the weather data is loading fine. @quequotion You need to update the dependency for AUR to point to this version of libgweather.

**Comment by casasfernando** [comment]
@DeeeeLAN I'm indeed using libgweather 3.36.1 which is the current version in elementary OS 6.x.
And you are correct you need to build against that version and it works just fine.
I will update the code once elementary OS 7 is released, which is based in Ubuntu 22.04 and will likely use a newer libgweather library that supports setting the contact info.

As a general rule of tumb and as mentioned in https://github.com/casasfernando/wingpanel-indicator-weather/issues/8#issuecomment-1116696571 only elementary OS is officially tested and supported so whenever building the indicator for any other distro, always make sure if the issue is still happening when building against the same dependencies versions as the ones in elementary OS.

I'm closing this issue.
